src/dune_rules/pkg_config.ml

```diff
@@ -20,8 +20,14 @@ module Query = struct
     | Libs s -> sprintf "%s.libs" s
     | Cflags s -> sprintf "%s.cflags" s
 
-  let to_args t : _ Command.Args.t list =
+  let to_args t ~env : _ Command.Args.t list =
+    let env_args : _ Command.Args.t list =
+      match Env.get env "PKG_CONFIG_ARGN" with
+      | Some s -> [ As (String.split_on_char ~sep:' ' s) ]
+      | None -> []
```

Comment:
You can use Arg.empty here and remove the S later. This doesn't need to be a list AFAICT.